### ISSUE TITLE
JS validation errors not accessible to screen readers

### ISSUE THREAD

**Issue by briri** [DMPRoadmap/roadmap#1446]
The JS `ariatiseForm.js` and `validation.js` error messages are not announced to screen readers.

Recommendation is to add a screen reader only alert at the top of the page in the current notification section that: A) lets the user know that some fields had errors, B) lets them know which fields, C) sets the focus back to the 1st field that failed validation.

This feedback was provided from the UCOP electronic accessibility review. It was defined as having 'High' priority.

**Comment by briri** [comment]
Recently updated the JS to set the focus to the first alert (may be necessary to set the focus to the actual input element though for good accessibility).

**Comment by briri** [comment]
We need to reevaluate the ariatiseForm JS file. We added a more versatile validations JS file which is basically a copy of the ariatiseForm file but is more flexible. It allows the caller to enable/disable and trigger validation of a field on demand. The ariatiseForm code only allows the caller to initialize an entire form which adds handlers to the form submit.

The validations JS file should be updated to ensure:
- Proper handling of checkboxes, radio buttons and grouped controls (instances where at least one of the contained input elements is required). This would mean that the HTML could apply `aria-required` to the `fieldset`. 
- We should also allow the caller to pass in the position of the error (top, bottom, left, right).
- Make sure that the `validate` method is called on form submit as well as input blur

We should then switch over all of the JS to use the newer validations JS file.

**Comment by briri** [comment]
This should be resolved by allowing Rails to handle validation. Errors appear in the standard notification area now which has appropriate aria tags.
